src/compas/geometry/splits/splits.py

```diff
@@ -0,0 +1,113 @@
+from __future__ import print_function
+from __future__ import absolute_import
+from __future__ import division
+
+from compas.utilities import pairwise
+
+from compas.datastructures import Mesh
+from compas.geometry import Polyline
+from compas.geometry import intersection_segment_plane
+from compas.geometry import intersection_mesh_plane
+from compas.geometry import subtract_vectors
+from compas.geometry import dot_vectors
+
+__all__ = [
+    'split_polyline_plane',
+    'split_mesh_plane',
+]
+
+
+def split_polyline_plane(polyline, plane):
+    """Split a polyline by a plane, returns a list of polylines, if there has been intersections
+    Returns list of the splitted polylines
+
+    Parameters
+    ----------
+    polyline : compas.datastructures.Polyline
+    plane : compas.geometry.Plane
+
+    Returns
+    -------
+    polylines : list of compas.geometry.Polyline
+    """
+    splitted_polylines_points = []
+    sublist = []
+    for i, segment in enumerate(pairwise(polyline)):
+        sublist.append(segment.start)
+        temp_intersection = intersection_segment_plane(segment, plane)
+        if temp_intersection:
+            sublist.append(temp_intersection)
+            splitted_polylines_points.append(sublist)
+            sublist = []
+            sublist.append(temp_intersection)
+        if i == len(pairwise(polyline))-1:
+            sublist.append(segment.end)
+            splitted_polylines_points.append(sublist)
+
+    return [Polyline(sublist) for sublist in splitted_polylines_points]
+
+
+def split_mesh_plane(mesh, plane, open=True):  # compas
+    """ Calculate all the intersections between edges of the mesh and cutting plane,
+    and splits every mesh edge at the intersection point, if it exists.
+    Returns a list of the resulting splitted meshes.
+
+    Parameters
+    ----------
+    mesh : compas.datastructures.Mesh
+    plane : compas.geometry.Plane
+
+    Returns
+    -------
+    splitted_meshes : list of compas.datastructures.Mesh
+    """
+
+    def mesh_from_split(self, mesh, v_keys, f_keys, intersections, open=True):  # compas
+        """ Return a mesh from the positive or negative verts, faces and the intersection verts of a splitted mesh
```

Comment:
There's something odd about this. Why is it a nested function at all? I would move it outside, if you don't want to expose it, that's fine, don't add it to the `__all__` and maybe also prefix it with `_` to make it clear is private, ie. `_mesh_from_split`. But then, there's the `self` here as first param that confusingly makes it look like a class-bound method.

Reply:
sorry, my mistake, i forgot to remove the self. Originally i created this function as a class function. Apart from that, i agree it makes sense to place it outside.